File: java/aula16-threadpools/src/main/java/pt/isel/pc/parallel/Stats.java
package pt.isel.pc.parallel;

import pt.isel.pc.utils.WorkerThreadReport;

import java.util.concurrent.atomic.AtomicInteger;

public class Stats {
	private final AtomicInteger workUnits = new AtomicInteger();


	public  Stats() {
		WorkerThreadReport.clear();
		workUnits.set(0);
	}

	public void collect() {
		//int used = WorkerThreadReport.createdThreads();
		workUnits.incrementAndGet();

		//System.out.printf("used= %d, wi=%d\n", used, wi);
		WorkerThreadReport.registerWorker();
	}


	public  void show() {
		System.out.println("Total of work units: " + workUnits);
		System.out.println("Total of worker threads: " + WorkerThreadReport.createdThreads());
	}

}

File: backup/dotnet/Aula16-threadpools/Stats.cs
using SynchUtils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Threading;

namespace Aula16_threadpools {
    class Stats {
        private volatile int workUnits;


        public Stats() {
            WorkerThreadReport.Clear();
            workUnits = 0;
        }

        public void Collect() {
            //int used = WorkerThreadReport.createdThreads();
            Interlocked.Increment(ref workUnits);

            //System.out.printf("used= %d, wi=%d\n", used, wi);
            WorkerThreadReport.RegisterWorker();
        }


        public void Show() {
            Console.WriteLine("Total of work units: {0}", workUnits);
            Console.WriteLine("Total of worker threads: {0}", WorkerThreadReport.CreatedThreads);
        }
    }
}
